webhook/psbinding/reconciler.go

```diff
@@ -232,36 +238,38 @@ func (r *BaseReconciler) RemoveFinalizer(ctx context.Context, fb kmeta.Accessor)
 func (r *BaseReconciler) labelNamespace(ctx context.Context, subject tracker.Reference) error {
 
 	// Determine the GroupVersionResource of the subject reference
-	gv, err := schema.ParseGroupVersion("v1")
-	if err != nil {
-		logging.FromContext(ctx).Infof("Error parsing GroupVersion %v: %v", "v1", err)
-		return err
+	gvr := schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "namespaces",
 	}
-	gvr := apis.KindToResource(gv.WithKind("Namespace"))
 
 	// Don't remove any existing labels or add ours more than once
-	unstructuredLabels, err := r.DynamicClient.Resource(gvr).Get(subject.Namespace, metav1.GetOptions{})
+	_, namespaceLister, err := r.Factory.Get(gvr)
 	if err != nil {
-		logging.FromContext(ctx).Infof("Error getting labels from namespace: %s: %v", subject.Namespace, err)
+		logging.FromContext(ctx).Infof("Error getting lister for namespaces: '%v': %v", gvr, err)
+		return err
 	}
-	if unstructuredLabels != nil {
-		labels := unstructuredLabels.GetLabels()
-		if labels["bindings.knative.dev/exclude"] != "" || labels["bindings.knative.dev/include"] != "" {
-			return nil
-		}
+	namespaceObject, err := namespaceLister.Get(subject.Namespace)
+	if apierrs.IsNotFound(err) {
+		logging.FromContext(ctx).Infof("")
+		return err
+	} else if err != nil {
+		return err
 	}
-	patch, err := json.Marshal(map[string]interface{}{
-		"metadata": map[string]interface{}{
-			"labels": map[string]string{"bindings.knative.dev/include": "true"},
-		},
-	})
+	labels := (namespaceObject.(*duckv1.WithPod)).GetLabels()
```

Comment:
This is sort of abusing the duck factory because it's not a WithPod.  I was thinking we'd setup a namespace informer in the way our other reconcilers do this and store the lister on the reconciler.